### PR TITLE
[BEAM-333][flink] make unbounded sources stoppable

### PR DESCRIPTION
**Pull request by mxm**

### CODE REVIEW

**Comment by aljoscha** [comment]
Nice and easy 👍 

By the way, there is also `BoundedSourceWrapper`, which could also be made stoppable.

**Comment by mxm** [comment]
+1 Changed that one as well.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java [hunk 1]
```diff
@@ -158,7 +159,7 @@ public void run(SourceContext<WindowedValue<OutputT>> ctx) throws Exception {
     }
 
     LOG.info("Unbounded Flink Source {}/{} is reading from sources: {}",
-        subtaskIndex,
+        subtaskIndex + 1,
```
Comment (aljoscha) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java]: Why did you add this? I know it's very confusing that "0/1" means first source of one but that's how it is in all of Flink with the subtask index. Maybe we should keep that for consistency.

**Comment by aljoscha** [comment]
Looks good once travis gives the go-ahead. I had one comment about a log statement.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java (hunk 1 above)
Comment (mxm) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java]: Yes, it is confusing. Either we start the index at 1 or we change the message to something like "Unbounded Flink Source with id {} ({} in total)".

**Pull request merged**